packages/next-polyfill-module/src/index.js

```diff
@@ -118,3 +118,17 @@ if (!Object.fromEntries) {
     }, {})
   }
 }
+
+/**
+ * Available in:
+ * Edge: 79
+ * Firefox: 65
+ * Chrome: 71
+ * Safari: 12.1
+ *
+ * https://caniuse.com/mdn-javascript_builtins_globalthis
+ */
+// Assume this polyfill module only running in browser environment.
+if (typeof globalThis === 'undefined' && typeof window !== 'undefined') {
+  window.globalThis = window
+}
```

Comment:
```suggestion
if (
  typeof globalThis !== 'object'
  // check corrupted globalThis
  // https://github.com/zloirock/core-js/blob/524af814dc4aef5d05102278ef288dfbd401ec55/packages/core-js/internals/global.js#L2
  || globalThis.Math !== Math
) {
  // use "self" as a fallback in case "window" doesn't exist
  const global = typeof window !== 'undefined' ? window : self;
  global.globalThis = global;
}
```

Reply:
To prevent `Function('return this')()`, you can also use another approach: https://mathiasbynens.be/notes/globalthis

```suggestion
if (typeof globalThis !== 'object') {
  Object.defineProperty(Object.prototype, '_T_', {
    configurable: true,
    get: function get() {
      var global = this || self;
      global.globalThis = global;
      delete Object.prototype._T_;
    }
  });
  _T_;
);

```